proto/executor.proto

```diff
@@ -135,16 +135,27 @@ message ANNQueryInfo {
 	optional VectorDistanceMetric distance_metric = 2 [(gogoproto.nullable) = false];
 	optional uint32 top_k = 3 [(gogoproto.nullable) = false];
 	optional string column_name = 4 [(gogoproto.nullable) = false]; // For debug purpose only. Currently only used in explain.
-	optional int64 column_id = 5 [(gogoproto.nullable) = false];
+	// deprecated field, we use column to get the id first, and this field will not be set again.
+	// Retain this field to be compatible with older versions of tidb
+	optional int64 deprecated_column_id = 5;
+	// Persists the original vector column's type information (including nullability)
+	// to ensure correct data handling. This field is always populated with the column's
+	// schema metadata, regardless of whether enable_distance_proj is enabled.
+	optional ColumnInfo column = 6 [(gogoproto.nullable) = false];
 
-	optional bytes ref_vec_f32 = 6; // The reference vector to calculate distance with, with each element is a Float32
-	reserved 7; // reserved for ref_vec_f64
+	optional bytes ref_vec_f32 = 7; // The reference vector to calculate distance with, with each element is a Float32
+	reserved 8; // reserved for ref_vec_f64
 
-	optional int64 index_id = 8 [(gogoproto.nullable) = false];
+	optional int64 index_id = 9 [(gogoproto.nullable) = false];
```

Comment:
Please review protobuf documentation about how to make changes:

https://protobuf.dev/programming-guides/proto3/#updating